parser/Cargo.toml

```diff
@@ -10,10 +10,10 @@ repository = "https://github.com/knurling-rs/defmt"
 version = "0.3.4"
 
 [dependencies]
-thiserror = "1.0"
+thiserror = "2"
 
 [dev-dependencies]
-rstest = { version = "0.19", default-features = false }
+rstest = { version = "0.23", default-features = false }
```

Comment:
Unless we rely on a feature in 0.23, we don't need this. Cargo will resolve to the latest anyway and there's no use on over-specifying the dependencies in case users have conflicting requirements.